### ISSUE TITLE
optimizing reduce function

### ISSUE THREAD

**Issue by maucon** [TheSwerik/Fraction#1]
**change**
```
    public Fraction reduce() {
        for (int i = 2; i <= Math.min(this.numerator, this.denominator); i++) {
            if (((double) this.numerator) / i == (int) ((double) this.numerator / i) &&
                    ((double) this.denominator) / i == (int) ((double) this.denominator / i)) {
                this.numerator /= i;
                this.denominator /= i;
                i = 2;
            }
        }
        return this;
    }
```
**to**
```
public Fraction reduce() {
       gcd = gcd(this.numerator, this.denominator);
       this.numerator /= gcd;
       this.denominator /= gcd;
       return this;
}
```

**Comment by TheSwerik** [comment]
fixed since [57b4a61](https://github.com/TheSwerik/Fraction/commit/57b4a61f6b43ec5c9ad275c850e34abb90b75463)

**Comment by maucon** [comment]
succ
